src/apps/answers/service.py

```diff
@@ -156,6 +157,14 @@ async def _validate_answer(self, applet_answer: AppletAnswerCreate):
             elif existed_answer.respondent_id != self.user_id:
                 raise WrongRespondentForAnswerGroup()
 
+        activity_history = await ActivityHistoriesCRUD(
+            self.session
+        ).get_by_activity_id(activity_id=applet_answer.activity_id)
```

Comment:
Here you need to check exact version of activity for which answer is sent.
So need to get by id_version generating it with activity_id and version
Check ActivityHistoriesCRUD.get_by_id

Reply:
fixed